src/interfaces/ISavingCircles.sol

```diff
@@ -24,6 +28,104 @@ interface ISavingCircles {
     uint256 maxDeposits;
   }
 
+  /**
+   * @notice Data structure for comprehensive user circle information
+   * @param circleId The ID of the circle
+   * @param circleInfo The circle information
+   * @param userBalance The user's current balance in the circle
+   * @param isMember Whether the user is a member of the circle
+   * @param isOwner Whether the user owns the circle
+   * @param isCurrentWithdrawer Whether the user is the current withdrawer
+   * @param canWithdraw Whether the user can currently withdraw
+   * @param nextWithdrawTime When the next withdrawal can occur
+   * @param depositWindowEnd When the current deposit window ends
+   * @param isExpired Whether the circle has expired
+   * @param isDecommissioned Whether the circle has been decommissioned
+   * @param totalPoolBalance Total balance of all members in the circle
+   * @param remainingDepositsNeeded Number of members who still need to complete deposits
+   * @param completedRounds Number of withdrawal rounds completed
+   * @param totalRounds Total number of withdrawal rounds
+   */
+  struct UserCircleData {
+    uint256 circleId;
+    Circle circleInfo;
+    uint256 userBalance;
+    bool isMember;
+    bool isOwner;
+    bool isCurrentWithdrawer;
+    bool canWithdraw;
+    uint256 nextWithdrawTime;
+    uint256 depositWindowEnd;
+    bool isExpired;
```

Comment:
keep bools together for better storage packing. also makes its slightly more readable imo